annsa/hyperparameter_models.py

```diff
@@ -0,0 +1,237 @@
+    """
+    number_layers = choice([1, 2, 3])
+    dense_nodes = 2**np.random.randint(5, 10, number_layers)
+    dense_nodes = np.sort(dense_nodes)
+    dense_nodes = np.flipud(dense_nodes)
+    model_features = dnn_model_features(
+        learining_rate=10**np.random.uniform(-4, -1),
+        l2_regularization_scale=10**np.random.uniform(-2, 0),
+        dropout_probability=np.random.uniform(0, 1),
+        batch_size=2**np.random.randint(4, 10),
+        output_size=all_keys_binarized.shape[1],
+        dense_nodes=dense_nodes,
+        activation_function=choice([tf.nn.tanh, tf.nn.relu]),
+        output_function=None,
+        scaler=choice(scaler_choices))
+
+    model = DNN(model_features)
+
+    return model, model_features
+
+
+def generate_random_cnn1d_architecture(cnn_filters_choices,
+                                       cnn_kernel_choices,
+                                       pool_size_choices):
+    """
+    Makes a random convolutional model features given some parameters.
+
+    Parameters:
+    -----------
+        cnn_filters_choices : list, int
+            List of number of filter to use for each convolutional layer.
+        cnn_kernel_choices : list, int
+            List of filter lengths to use for each convolutional layer.
+        pool_size_choices : list, int
+            List of pooling lengths to use for each convolutional layer.
+
+    Returns:
+    --------
+        model_features : class
+            Class that describes the structure of a the 1D convolutional
+            network
+    """
+
+    cnn_filters = choice(cnn_filters_choices)
+    cnn_kernel_choice = choice(cnn_kernel_choices)
+    pool_size_choice = choice(pool_size_choices)
+
+    cnn_kernel = cnn_kernel_choice*(len(cnn_filters))
+    cnn_strides = (1,)*(len(cnn_filters))
+    pool_size = pool_size_choice*(len(cnn_filters))
+    pool_strides = (2,)*(len(cnn_filters))
+
+    number_layers = np.random.randint(1, 4)
+    dense_nodes = (10**np.random.uniform(1,
+                                         np.log10(1024/(2**len(
+                                             cnn_filters))),
+                                         number_layers)).astype('int')
+    dense_nodes = np.sort(dense_nodes)
+    dense_nodes = np.flipud(dense_nodes)
+
+    model_features = cnn1d_model_features(
+            trainable=None,
+            learining_rate=None,
+            batch_size=None,
+            output_size=None,
+            scaler=None,
+            activation_function=None,
+            output_function=None,
+            Pooling=None,
+            l2_regularization_scale=None,
+            dropout_probability=None,
+            cnn_filters=cnn_filters,
+            cnn_kernel=cnn_kernel,
+            cnn_strides=cnn_strides,
+            pool_size=pool_size,
+            pool_strides=pool_strides,
+            dense_nodes=dense_nodes
+            )
+
+    return model_features
+
+
+def make_conv1d_model(all_keys_binarized):
```

Comment:
You already have all of these `make_model` functions defined in the tests. Would you rather have them defined here and then used by the tests?